### ISSUE TITLE
Invoking PushManager.dispatchInfoForChid on an unknown chid throws KotlinNullPointerException

### ISSUE THREAD

**Issue by jonalmeida** [mozilla/application-services#2698]
**TL;DR**, `PushManager.dispatchInfoForChid` returns a `DispatchInfo?` (nullable) but the function `DispatchInfo.fromMessage` returns a `DispatchInfo` (non-null), so when a null value is returned, a nullptr exception is thrown.

---

During my WebPush testing, a site can request a subscription _before_ making one (to avoid making new ones.

On GV this is [`WebPushDelegate.onGetSubscription`][0] where the site only provides the `scope` (usually their site address). In A-C, to satisfy this requirement we call the rust `PushManager.subscribe` to create a subscription or if one already exists, retrieve it from the on-device cache.

An issue can arise then where we have to call `subscribe` without have an `appServerKey` when invoked by `WebPushDelegate.onGetSubscription`. To avoid this, we try to call `PushManager.dispatchInfoForChid` to retrieve the `appServerKey` if one exists.

However, it seems the nullability of the `DispatchInfo` at the Kotlin layer in a-s seems incorrect, from what I can tell. The API says we can receive a [`DispatchInfo?`][1] (nullable), but the function that creates one [`DispatchInfo.fromMessage`][2] assumes a non-null value. This leads to a runtime error with a similar stack:

```
2020-02-25 00:26:21.063 30991-31086/org.mozilla.reference.browser.debug E/AndroidRuntime: FATAL EXCEPTION: pool-5-thread-1
    Process: org.mozilla.reference.browser.debug, PID: 30991
    kotlin.KotlinNullPointerException
        at mozilla.appservices.push.PushManager.dispatchInfoForChid(PushManager.kt:499)
        at mozilla.components.feature.push.RustPushConnection.subscribe(Connection.kt:132)
        at mozilla.components.feature.push.AutoPushFeature$subscribe$3$2.invokeSuspend(AutoPushFeature.kt:206)
        at mozilla.components.feature.push.AutoPushFeature$subscribe$3$2.invoke(Unknown Source:10)
        at mozilla.components.feature.push.AutoPushFeatureKt$launchAndTry$1.invokeSuspend(AutoPushFeature.kt:356)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

[0]: https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/WebPushDelegate.html#onGetSubscription-java.lang.String-
[1]: https://github.com/mozilla/application-services/blob/master/components/push/android/src/main/java/mozilla/appservices/push/PushManager.kt#L127
[2]: https://github.com/mozilla/application-services/blob/master/components/push/android/src/main/java/mozilla/appservices/push/PushManager.kt#L264

┆Issue is synchronized with this [Jira Bug](https://jira.mozilla.com/browse/SYNC-862)
┆Story Points: 3
┆Sprint: SYNC - end 2020-02-28
